.github/workflows/trigger-ci.yml

```diff
@@ -0,0 +1,153 @@
+name: Trigger CI for External PRs
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  contents: write
+  pull-requests: write
+  issues: write
+
+jobs:
+  trigger-ci:
+    name: Trigger CI Run
+    # Only run on PR comments
+    if: github.event.issue.pull_request && contains(github.event.comment.body, '/run-ci')
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Check if commenter has admin permissions
+        id: check-permissions
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script: |
+            try {
```

Comment:
I'd prefer we pull these out into script files, like [here](https://github.com/vercel/vercel/blob/main/.github/workflows/update-remix-run-dev.yml#L28-L30).